From f317ec41a1948e35330364c3120d937cc9934888 Mon Sep 17 00:00:00 2001
From: Elliot Bobrow <77182873+ebobrow@users.noreply.github.com>
Date: Mon, 20 Jun 2022 12:18:05 -0700
Subject: [PATCH] LaTeX: Handle formatted text inside code inline (#8129)

Add `formatCode` function to Text.Pandoc.Shared [API change].

Use this in the LaTeX reader so that e.g.
`\texttt{\textbf{bold code}}` is parsed as `Strong [Code ("",[],[]) "bold code"]`.
---
 src/Text/Pandoc/Readers/LaTeX.hs | 11 ++++-------
 src/Text/Pandoc/Shared.hs        | 23 ++++++++++++++++++++++-
 test/command/7525.md             | 19 +++++++++++++++++++
 3 files changed, 45 insertions(+), 8 deletions(-)
 create mode 100644 test/command/7525.md

diff --git a/test/command/7525.md b/test/command/7525.md
new file mode 100644
index 000000000..788c749a6
--- /dev/null
+++ b/test/command/7525.md
@@ -0,0 +1,19 @@
+```
+% pandoc -f latex -t native
+\texttt{Normal code. \emph{Emph and code.} \textsc{\textbf{Bold small caps.}} \sout{Strikeout. \underline{Strikeout and underline.}}}
+^D
+[ Para
+    [ Code ( "" , [] , [] ) "Normal code. "
+    , Emph [ Code ( "" , [] , [] ) "Emph and code." ]
+    , Code ( "" , [] , [] ) " "
+    , SmallCaps
+        [ Strong [ Code ( "" , [] , [] ) "Bold small caps." ] ]
+    , Code ( "" , [] , [] ) " "
+    , Strikeout
+        [ Code ( "" , [] , [] ) "Strikeout. "
+        , Underline
+            [ Code ( "" , [] , [] ) "Strikeout and underline." ]
+        ]
+    ]
+]
+```
-- 
2.25.1

